gh-actions/go/generate/action.yaml

```diff
@@ -52,10 +52,21 @@ runs:
       with:
         token: ${{ inputs.token }}
     - name: Install latest protoc
-      uses: arduino/setup-protoc@v3
       if: ${{ steps.proto-deps.outputs.needs-protoc == 'true' }}
-      with:
-        repo-token: ${{ inputs.token }}
+      run: |
+        AUTH_HEADER=""
+        if [ -n "${{ inputs.token }}" ]; then
+          AUTH_HEADER="-H 'Authorization: Bearer ${{ inputs.token }}'"
+        fi
+        PROTOC_VERSION=$(eval curl -s $AUTH_HEADER \
```

Comment:
I don’t think you did that without research, but why do we need the AUTH_HEADER? What happens if it’s empty as there is one code path without it?

General thing is that I would prefer we avoid requiring it (for forks in particular) if possible. This was my main grip with the arduino action: we needed to pass it.

Reply:
Using the GitHub token in the request to the GitHub API endpoint avoids rate limiting. The `arduino/protoc` action used it for the same purpose, and it did not actually require it, but we did recently pass it with 895c1139e7452498a8155c5e352843a6eb307faa. I don't want to break things for anyone, so I think its better to keep using it.

Reply:
Indeed, I can see that we can hit the limit quite easily, thanks for looking at this! It makes sense to keep this logic then.

Maybe just a warning/notice to be printed in the GH summary in case the download fails and we have an empty `AUTH_HEADER` with instructions to pass the token? That would help debugging.

Reply:
Okay, see you on https://github.com/canonical/desktop-engineering/pull/81 :)

Reply:
done, please take a look